frontend/vite-project/src/stores/paymentStore.ts

```diff
@@ -0,0 +1,21 @@
+import { create } from 'zustand';
+import { PaymentFormTypes, PaymentState } from '../types/paymentFormTypes';
+
+export const usePaymentStore = create<PaymentState>((set) => ({
+    formData: {
+        cardholderName: '',
+        cardNumber: '',
+        expMonth: '',
+        expYear: '',
+        cvc: '',
+        saveCard: false,
+    },
+    setFormData: (data) => set({ formData: data }),
+    updateFormField: (key, value) =>
+        set((state) => ({
+            formData: {
+                ...state.formData,
+                [key]: value,
+            },
+        })),
```

Comment:
```suggestion
        })),
 export const useFormData = () => usePaymentStore((state) => state.formData);
 export const usePaymentActions = () => usePaymentStore((state) => state.actions)
```
Export custom hooks for cleaner and more modular access to the shipping store
useFormData returns the current form data state.
usePaymentActions provides access to all state-modifying actions.